packages/truffle-db/src/artifacts/schema.ts

```diff
@@ -75,5 +96,61 @@ export const schema = mergeSchemas({
       },
       
     },
+
+    Contract: {
+      name: {
+        fragment: "... on ContractObject { name: contractName }"
+      }, 
+      sourceContract: {
+        fragment: 
+        `... on ContractObject { 
+          ast
+        }`,
+        resolve: (obj, ast) => {
+          const { name, source } = obj;
+          const sourceContract = {
+            name: name, 
+            source: source, 
+            ast: ast
+          }
+          return sourceContract;
+        }
+      }, 
+    },
+
+    ContractConstructor: {
+      contract: {
+        fragment: 
+        `... on ContractObject { 
+          contents,
+          sourcePath,
```

Comment:
`contents` and `sourcePath` need to get grouped into `source`

Reply:
Oh we also probably need to add `name: contractName` as well, since it's read from `obj`.